.github/actions/nexus-associate-tag/action.yml

```diff
@@ -0,0 +1,42 @@
+name: Associate tag with artifacts
+description: Associate tag with artifacts in Nexus
+inputs:
+  tag-name:
+    description: The name of the tag
+    required: true
+  nexus-username:
+    description: Nexus user name
+    required: true
+  nexus-password:
+    description: Nexus password
+    required: true
+  nexus-url:
+    description: Base URL to the Nexus server
+    required: false
+    default: "https://artifacts.alfresco.com/nexus"
+  group-id:
+    description: The maven group-id of the component
+    required: true
+  version:
+    description: The version of the component
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: associate-tag
+      id: associate-tag
+      shell: bash
+      env:
+        TAG_NAME: ${{ inputs.tag-name }}
+        NEXUS_USERNAME: ${{ inputs.nexus-username }}
+        NEXUS_PASSWORD: ${{ inputs.nexus-password }}
+        NEXUS_URL: ${{ inputs.nexus-url }}
+        GROUP-ID: ${{ inputs.group }}
+        VERSION: ${{ inputs.version }}
+      run: |
+          echo "Associating tag $TAG_NAME with component $GROUP:$VERSION"
+            curl -u "${NEXUS_USERNAME}":"${NEXUS_PASSWORD}" \
+                -X POST "${NEXUS_URL}/service/rest/v1/tags/associate/${TAG_NAME}&group=${GROUP-ID}&version=${VERSION}" \
+                -H "Content-Type:application/json" \
+                -d '{}'
```

Comment:
Can't the group and version be passed here instead? (allows proper encoding of query parameters)

Reply:
Sorry, @atchertchian, I don't get what you mean here. Group and version are already passed as inputs and mapped to env vars

Reply:
It's a detail: query params can be passed separately so that they can be properly encoded, but that does not seem valid for POST after a quick try, and seems overkill given potential values for params, so nevermind